README.md

```diff
@@ -205,15 +205,42 @@ easy to fork and contribute any changes back upstream.
         cd ~/.pyenv && src/configure && make -C src
 
 2. **Define environment variable `PYENV_ROOT`** to point to the path where
-   pyenv repo is cloned and add `$PYENV_ROOT/bin` to your `$PATH` for access
-   to the `pyenv` command-line utility.
+   pyenv repo is cloned, add `$PYENV_ROOT/bin` to your `$PATH` for access
+   to the `pyenv` command-line utility, enable shims to be able to use Pyenv.
 
-   - For **bash**/**Zsh**:
-     ~~~ bash
-     echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.profile
-     echo 'export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.profile
-     echo 'eval "$(pyenv init --path)"' >> ~/.profile
-     ~~~
+   - For **bash**:
+
+      - **If your `~/.profile` sources `~/.bashrc` (Ubuntu):**
+         <!--This way is an alternative option and needn't be replicated to `pyenv init`-->
+         ~~~ bash
+         echo -e 'if shopt -q login_shell; then' \
+               '\n  export PYENV_ROOT="$HOME/.pyenv"' \
+               '\n  export PATH="$PYENV_ROOT/bin:$PATH"' \
+               '\n eval "$(pyenv init --path)"' \
+               '\nfi' >> ~/.bashrc
+         ~~~
```

Comment:
This `if` block solution is not working for me on Ubuntu 20.04 with bash. This is what I have in `.bashrc`:
```shell
if shopt -q login_shell; then
  export PYENV_ROOT="$HOME/.pyenv"
  export PATH="$PYENV_ROOT/bin:$PATH"
  eval "$(pyenv init --path)"
fi
```
If I echo my path I get:
```shell
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
```

Reply:
@julianfortune does your `.profile` contain a line that sources `.bashrc`?

Reply:
@agoose77 Yes, [this](https://gist.github.com/julianfortune/27fc6218b86dfb55b5495045f93c07c1) is what is contained in .profile by default.

Reply:
Okay, the `~/.bashrc` appending is not good enough, we need to come up with something better.
I'll split this off into a seprate PR and dissosiate this one from https://github.com/pyenv/pyenv/issues/1914.

Reply:
What about a "C library ifndef"-style approach, something like:
```
if [ -z $PYENV_PATH_SET ]; then
  export PYENV_ROOT="$HOME/.pyenv" 
  export PATH="$PYENV_ROOT/bin:$PATH" 
  eval "$(pyenv init --path)"
  export PYENV_PATH_SET='true'
fi

Reply:
@julianfortune yes, when you SSH with no commands you are given an interactive login shell, so your `.profile` is sourced (or other profile-like file) (see [here](https://web.archive.org/web/20121025183138/http://meta.ath0.com/2007/10/23/cleaning-up-bash-customizations/) or [here](http://www.solipsys.co.uk/new/BashInitialisationFiles.html) for more).

When you log in via the GUI, it varies per configuration. On Ubuntu with GDM only your `.profile` is explicitly sourced because `GDM` explicitly invokes `/bin/sh` (even if you change your default shell), and is launched as a login shell. 

So, for maximum compatibility, you end up wanting to have a `.bash_profile` that loads `.profile` and `.bashrc`, and then a much smaller `.profile` for just doing sh+bash compatible things.

Reply:
The problem is, `~/.bashrc` in included when `~/.profile` is executed as `bash` (by a Bash login shell) -- but is _not_ included when it's run as `sh` (by a GUI manager).

My current ideas are:

1. Appends blocks to both `~/.bashrc` and `~/.profile` so that the first is executed in Bash mode and the 2nd one (below) in non-Bash mode:

    ```bash
    if [ -z "$BASH_VERSION" ]; then
      export PYENV_ROOT="$HOME/.pyenv" 
      export PATH="$PYENV_ROOT/bin:$PATH" 
      eval "$(pyenv init --path)"
      export PYENV_PATH_SET='true'
    fi
    ```
    This has a downside of duplicated code. But this is the only solution achieveable with just appending.

2. Make an editing script (probably in Awk or Perl) that would insert the lines before the `if` block that sources `~/.bashrc`. Detect the necessary line with heuristics (e.g. a regex like `(^|\s)if\s.*\W(BASH_\w+|bash)\W`); if nothing found, append to the end.

    This will obviously break in environments with a very different `~/.profile` logic, and it'll be a rather complex command.